### PR TITLE
feat: add holdToDrag option

### PR DESCRIPTION
**Pull request by altschuler**
Add a `holdToDrag` option that delays the drag action on touch
devices, which is useful for items that should be both scrollable and
draggable.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1529,6 +1535,44 @@ function determineDragOperation(effectAllowed:string, dropEffect:string):string
...
+function onEvt(el: EventTarget, event: string, handler: () => any, context: any) {
+    if (context) {
+        handler = handler.bind(context);
+    }
+
+    el.addEventListener(event, handler, supportsPassive ? { passive: false } : false);
+
+    return {
+        off() {
+            return el.removeEventListener(event, handler, supportsPassive ? { passive: false } : false);
```
Comment (reppners) [inline src/index.ts]: This won't compile with TypeScript 2.3.3 as the signature for removeEventListener is not exactly the same as for addEventListener - maybe it's a bug in lib.d.ts.

Change to ` return el.removeEventListener(event, handler);` will make it compile. You can check by running `grunt` in your local checkout.

File: src/index.ts [hunk 2]
```diff
@@ -1529,6 +1535,44 @@ function determineDragOperation(effectAllowed:string, dropEffect:string):string
...
+    const cancel = onEvt(el, 'touchcancel', onReleasedItem, this);
+    const scroll = onEvt(window, 'scroll', onReleasedItem, this);
+}
+
+function onEvt(el: EventTarget, event: string, handler: () => any, context: any) {
+    if (context) {
+        handler = handler.bind(context);
+    }
+
+    el.addEventListener(event, handler, supportsPassive ? { passive: false } : false);
```
Comment (reppners) [inline src/index.ts]: No need to actually specify `passive: false` here as the event handlers registered with this method are passive. Pls change to `el.addEventListener(event, handler);`

Comment (altschuler) [reply src/index.ts]: Fixed

File: src/index.ts (hunk 1 above)
Comment (altschuler) [reply src/index.ts]: I got the error locally yeah, fixed :)

**Comment by reppners** [comment]
Thx! Will cut a release soon :)

**Pull request merged**